crates/bevy_pbr/src/render/mesh.rs

```diff
@@ -620,6 +630,15 @@ impl SpecializedMeshPipeline for MeshPipeline {
             depth_write_enabled = true;
         }
 
+        if key.contains(MeshPipelineKey::TONEMAP_IN_SHADER) {
+            shader_defs.push("TONEMAP_IN_SHADER".to_string());
+        }
+
+        let format = match key.contains(MeshPipelineKey::HDR) {
+            true => ViewTarget::TEXTURE_FORMAT_HDR,
+            false => TextureFormat::bevy_default(),
```

Comment:
The ‘otherwise’ there was about in the low dynamic range case that it shouldn’t be bevy_default().

We will for one probably need to get rid of bevy_default() as we need to detect what the preferred swapchain formats are and then choose one of those instead of hard-coding. And we don’t need to use the same format for rendering as for the swapchain, unless we’re rendering into the swapchain. We may later need to detect if there are any non-srgb formats support by the swapchain and see what behaviours are like for displaying those. The purpose would be to have more control over gamma by applying it manually in the final steps, but I don’t know if OSes/drivers do something like “oh you chose a non-srgb format, let me apply srgb gamma correction for you.”

So I am proposing that we use Rgba8Unorm for LDR main pass output, and I think this PR ises Rgba16Float for HDR, and in a separate PR that we detect what the preferred swapchain formats are and choose one.